src/python/pants/option/arg_splitter.py

```diff
@@ -273,15 +272,23 @@ def _descope_flag(self, flag: str, default_scope: str) -> tuple[str, str]:
         return default_scope, flag
 
     def _at_flag(self) -> bool:
-        return (
-            bool(self._unconsumed_args)
-            and self._unconsumed_args[-1].startswith("-")
-            and not self._at_double_dash()
-            and not self._at_scope()
-        )
+        if not self._unconsumed_args:
+            return False
+        arg = self._unconsumed_args[-1]
+        if not arg.startswith("--") and not self.is_level_short_arg(arg):
+            return False
+        return not self._at_standalone_double_dash() and not self._at_scope()
 
     def _at_scope(self) -> bool:
         return bool(self._unconsumed_args) and self._unconsumed_args[-1] in self._known_scopes
 
-    def _at_double_dash(self) -> bool:
+    def _at_standalone_double_dash(self) -> bool:
+        """At the value `--`, used to start passthrough args."""
         return bool(self._unconsumed_args) and self._unconsumed_args[-1] == "--"
+
+    def is_level_short_arg(self, arg: str) -> bool:
+        """We special case the `--level` global option to also be recognized with `-l`.
+
+        It's important that that this be classified as a global option.
```

Comment:
that that this?

Reply:
Oops. I have a followup coming in a few minutes and will fix there